### PR TITLE
Fix two factor authentication notification

### PR DESCRIPTION
**Pull request by camilasan**
Fix for #4841

### CODE REVIEW

**Comment by jancborchardt** [comment]
@camilasan nice! :) Do you mind posting a screenshot for a much quicker design review? Thank you!

**Review by jancborchardt — APPROVED** [review]
If it’s the screenshot from https://github.com/nextcloud/desktop/issues/4841#issuecomment-1252767634, looks good! :)

The message does mention "Approve or deny" but the buttons are only "Approve" or "Cancel", so maybe they should be named accordingly?

File: src/gui/tray/ActivityItemActions.qml [hunk 1]
```diff
@@ -46,10 +46,11 @@ RowLayout {
             imageSource: model.modelData.imageSource ? model.modelData.imageSource + root.adjustedHeaderColor : ""
             imageSourceHover: model.modelData.imageSourceHovered ? model.modelData.imageSourceHovered + UserModel.currentUser.headerTextColor : ""
 
-            textColor: imageSource !== "" ? root.adjustedHeaderColor : Style.ncTextColor
-            textColorHovered: imageSource !== "" ? UserModel.currentUser.headerTextColor : Style.ncTextColor
+            textColor: imageSource !== "" || primary ? root.adjustedHeaderColor : Style.ncTextColor
+            textColorHovered: imageSource !== "" || primary ? UserModel.currentUser.headerTextColor : Style.ncTextColor
 
             bold: primary
```
Comment (claucambra) [inline src/gui/tray/ActivityItemActions.qml:52]: IMO this should now be decided internally by the `ActivityActionButton`, where `property bool bold: primaryButton`, as we never set it to anything different here

File: src/gui/tray/ActivityItemActions.qml [hunk 2]
```diff
@@ -46,10 +46,11 @@ RowLayout {
             imageSource: model.modelData.imageSource ? model.modelData.imageSource + root.adjustedHeaderColor : ""
             imageSourceHover: model.modelData.imageSourceHovered ? model.modelData.imageSourceHovered + UserModel.currentUser.headerTextColor : ""
 
-            textColor: imageSource !== "" ? root.adjustedHeaderColor : Style.ncTextColor
-            textColorHovered: imageSource !== "" ? UserModel.currentUser.headerTextColor : Style.ncTextColor
+            textColor: imageSource !== "" || primary ? root.adjustedHeaderColor : Style.ncTextColor
+            textColorHovered: imageSource !== "" || primary ? UserModel.currentUser.headerTextColor : Style.ncTextColor
```
Comment (claucambra) [inline src/gui/tray/ActivityItemActions.qml:50]: Throughout `ActivityItemActions` we are now basically always using the `CustomButton` for a situation where we have a primary button (before we would use `primary` to ensure we bolded the first button in the `CustomTextButton`)

Now that we always set things with `imageSource` and `primary` together, can we not just integrate `imageSource !== ""` into the `primary` property?

**Review by allexzander — APPROVED** [review]

File: src/gui/tray/ActivityItemActions.qml [hunk 3]
```diff
@@ -46,10 +46,10 @@ RowLayout {
             imageSource: model.modelData.imageSource ? model.modelData.imageSource + root.adjustedHeaderColor : ""
             imageSourceHover: model.modelData.imageSourceHovered ? model.modelData.imageSourceHovered + UserModel.currentUser.headerTextColor : ""
 
-            textColor: imageSource !== "" ? root.adjustedHeaderColor : Style.ncTextColor
-            textColorHovered: imageSource !== "" ? UserModel.currentUser.headerTextColor : Style.ncTextColor
+            textColor:  primary ? root.adjustedHeaderColor : Style.ncTextColor
```
Comment (claucambra) [inline src/gui/tray/ActivityItemActions.qml:49]:
```suggestion
            textColor: primary ? root.adjustedHeaderColor : Style.ncTextColor
```

**Review by claucambra — CHANGES_REQUESTED** [review]
A couple of small details

**Comment by claucambra** [comment]
@camilasan It seems to work great for most notifications except missed call notifications, where we now have two primary buttons:

Screenshot 2022-09-26 at 18 22 09

I am guessing this is not intentional?

**Comment by jancborchardt** [comment]
At some point, "Reply" would also be good to have in call notifications for server. cc @nickvergessen 

In the case of call notifications, "Call back" would be primary, "Reply" secondary.
For messages, "Reply" should stay primary of course.

**Comment by nickvergessen** [comment]
> At some point, "Reply" would also be good to have in call notifications for server.

https://github.com/nextcloud/notifications/issues/637

**Review by claucambra — APPROVED** [review]

**Comment by claucambra** [comment]
looks good now

**Pull request merged**